src/mesh/region.cc

```diff
@@ -568,3 +407,51 @@ Region RegionDB::find_by_dimid(DimIDIter it_id, unsigned int id, const std::stri
 
     return r_id;
 }
+
+void RegionDB::print_region_table(ostream& stream) const {
+	ASSERT(closed_, "RegionDB not closed yet.\n");
+
+	// print header
+	stream << endl << "----------- Table of all regions: -----------" << endl;
+	stream << std::setfill(' ') << setw(6) << "id" << " dim label" << setw(12) << "" << "contains regions" << endl;
+	// print data
+	for (RegionSetTable::const_iterator it = sets_.begin(); it != sets_.end(); ++it) { // iterates through RegionSets
+		unsigned int reg_id = RegionIdx::undefined;
+		string rset_label = it->first;
+		LabelIter label_it = region_table_.get<Label>().find(rset_label);
+		if ( label_it != region_table_.get<Label>().end() ) { // checks if Region with same label as RegionSet exists
```

Comment:
This condition is supposed to give same result as the latter one (it->second.size() ==1. Rather merge both conditions.
Print single region set by line, reporting labels of elementary regions for size>1, and ID  and dim for size==1.